### PR TITLE
indexer json rpc tests: checkpoint apis

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

title

## Test plan 

cargo test --package sui-indexer --test read_api_tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by emmazzz — APPROVED** [review]

File: crates/sui-indexer/src/test_utils.rs [hunk 1]
```diff
@@ -81,7 +86,7 @@ pub async fn start_test_indexer_impl(
     // Reduce the connection pool size to 10 for testing
     // to prevent maxing out
     let pool_config = ConnectionPoolConfig {
-        pool_size: 5,
+        pool_size: 10,
```
Comment (bmwill) [inline crates/sui-indexer/src/test_utils.rs:89]: can we keep these at a lower number?

File: crates/sui-indexer/tests/read_api_tests.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+use simulacrum::Simulacrum;
+use std::sync::Arc;
+use sui_indexer::apis::read_api::ReadApi;
+use sui_indexer::indexer_reader::IndexerReader;
+use sui_indexer::test_utils::{set_up, wait_for_checkpoint};
+use sui_json_rpc_api::ReadApiServer;
+use tempfile::tempdir;
+
+#[tokio::test]
+async fn test_read_api() -> RpcResult<()> {
```
Comment (bmwill) [inline crates/sui-indexer/tests/read_api_tests.rs:14]: Can we rename this test to be more descriptive that its testing the checkpoint functionality of the ReadApi and not any other apis?

File: crates/sui-indexer/src/test_utils.rs (hunk 1 above)
Comment (gegaowp) [reply crates/sui-indexer/src/test_utils.rs:89]: I increased this b/c the conn pool is shared by both writers and readers now, also the comment said 10, I am okay if there is strong reason to keep it 5 though

Comment (bmwill) [reply crates/sui-indexer/src/test_utils.rs:89]: yeah lets keep it at 5, this just reduces the amount of resources a test will actually need

**Pull request merged**